.github/workflows/default.yml

```diff
@@ -38,12 +40,8 @@ jobs:
 
       - uses: actions/setup-java@v3
         with:
-          distribution: 'temurin'
-          java-version: |
-            8
-            11
-            17
-            20
+          distribution: ${{ matrix.distribution }}
+          java-version: ${{ matrix.java-version }}
```

Comment:
🐛 
This doesn't look valid. It will install just a single java version and will rely on [foojay-toolchains](https://github.com/gradle/foojay-toolchains) to download needed distrubutions (during Gradle invocation). 
The build requires multiple java versions installed to run tests (see https://jakewharton.com/build-on-latest-java-test-through-lowest-java/ for more context)

Reply:
Your answer is complete and correct, thank you